src/SignalR/common/Shared/MessageBuffer.cs

```diff
@@ -181,19 +180,11 @@ public void Ack(AckMessage ackMessage)
 
     internal bool ShouldProcessMessage(HubMessage message)
     {
-        // TODO: if we're expecting a sequence message but get here should we error or ignore or maybe even continue to process them?
-        if (_waitForSequenceMessage)
+        // Technically handled by the 'is not HubInvocationMessage' check, but this is future proofing in case that check changes
+        // SequenceMessage should not be counted towards ackable messages
+        if (message is SequenceMessage)
         {
-            if (message is SequenceMessage)
-            {
-                _waitForSequenceMessage = false;
-                return true;
-            }
-            else
-            {
-                // ignore messages received while waiting for sequence message
-                return false;
-            }
+            return true;
```

Comment:
```suggestion
        if (message is SequenceMessage sequenceMessage)
        {
            // TODO: is a sequence message expected right now?
            if (sequenceMessage.SequenceId > _currentReceivingSequenceId)
            {
                throw new InvalidOperationException("Sequence ID greater than amount of messages we've received.");
            }

            _currentReceivingSequenceId = sequenceMessage.SequenceId;

            // Technically handled by the 'is not HubInvocationMessage' check, but this is future proofing in case that check changes
            // SequenceMessage should not be counted towards ackable messages
            return true;
```

And then just delete `ResetSequence`. Then you don't need the "// Safe to modify outside a lock because the only other modifier is ResetSequence()" comment. This isn't just the only other modifier, it's the only other reader or modifier, right?